File: app/src/main/java/com/example/bindechexclock/ui/romans/RomansViewModel.java
package com.example.bindechexclock.ui.romans;

import androidx.lifecycle.LiveData; // Changed
import androidx.lifecycle.MutableLiveData; // Changed
import androidx.lifecycle.ViewModel; // Changed

public class RomansViewModel extends ViewModel { // Corrected class declaration, was ViewModel{

    private MutableLiveData<String> mText;

    public RomansViewModel() {
        mText = new MutableLiveData<>();
        // mText.setValue("This is romans fragment"); // Default text removed
    }

    public LiveData<String> getText() {
        return mText;
    }

    public void setmText(String s) {
        mText.setValue(s);
    }
}

File: app/src/main/java/com/example/bindechexclock/ui/romans/RomansViewModel.kt
package com.example.bindechexclock.ui.romans

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class RomansViewModel : ViewModel() {

    // Private MutableLiveData that can be modified within the ViewModel.
    private val _text = MutableLiveData<String>()

    // Public LiveData that is exposed to the UI (Fragment) for observation.
    val text: LiveData<String> get() = _text

    // The original Java constructor initialized mText = new MutableLiveData<>()
    // In Kotlin, this is handled by the property initializer above.

    /**
     * Updates the text LiveData.
     * Called by RomansFragment when it receives updates from TimeManager.
     */
    fun updateText(newText: String) {
        _text.value = newText
    }
}
